ale_linters/go/gobuild.vim

```diff
@@ -1,17 +1,126 @@
-" Author: dzhou121 <dzhou121@gmail.com>
+" Author: Joshua Rubin <joshua@rubixconsulting.com>
 " Description: go build for Go files
 
-function! s:FindGobuildScript() abort
-    return g:ale#util#stdin_wrapper . ' .go go build -o /dev/null'
+" inspired by work from dzhou121 <dzhou121@gmail.com>
+
+function! s:goenv()
```

Comment:
Could you name your functions in PascalCase, per the Google style guide? https://google.github.io/styleguide/vimscriptguide.xml?showone=Naming#Naming